### PR TITLE
feat(zBar): add zBar barcode scanner plugin

### PR DESCRIPTION
**Pull request by AndreasGassmann**

### CODE REVIEW

File: src/plugins/z-bar.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+export class ZBar {
+
+  /**
+   * Open the scanner
+   * @param options { ZBarOptions } Scan options
+   * @param success { ZBarSuccessCallback } Success callback
+   * @param failure { ZBarErrorCallback } Error callback
+   */
+  @Cordova()
+  static scan(options: ZBarOptions, success: ZBarSuccessCallback, failure: ZBarErrorCallback): void { return; }
```
Comment (ihadeed) [inline src/plugins/z-bar.ts]: No need for successCallback and errorCallback here. 
Using the `@Cordova()` decorator with no options will automatically add these callbacks at the end of the arguments list and link them to a promise. See developer guide for more details.

So the correct line here would be:
```
static scan(options: ZBarOptions): Promise<any> { return; }
```

File: src/plugins/z-bar.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+}
+
+export interface ZBarSuccessCallback { (scanResult: string): void; }
```
Comment (ihadeed) [inline src/plugins/z-bar.ts]: Remove this.

File: src/plugins/z-bar.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+export interface ZBarErrorCallback { (error: string): void; }
```
Comment (ihadeed) [inline src/plugins/z-bar.ts]: Remove this.

File: src/plugins/z-bar.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+export interface ZBarOptions {
```
Comment (ihadeed) [inline src/plugins/z-bar.ts]: Document what each property represents here (use the information you have in the `@advanced` tag)

File: src/plugins/z-bar.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+ * @usage
+ * ```
+ * import { ZBar } from 'ionic-native';
+ *
+ * let zBarOptions = {
+ *       flash: "off",
+ *       drawSight: false
+ *     };
+ *
+ * ZBar.scan(zBarOptions, result => {
```
Comment (ihadeed) [inline src/plugins/z-bar.ts]: After fixing the method below (by returning a promise), edit the usage to something like:
```
ZBar.scan(zBarOptions).then(result => { ... }).catch(error => { ... }); 
```

**Comment by AndreasGassmann** [comment]
@ihadeed Thanks for your help.

I applied your changes, but it looks like I somehow messed up the merge. If my changes are ok, I can delete my fork/pull request, create a new one and add my changes there.

**Comment by ihadeed** [comment]
@AndreasGassmann yes, please do create a new clean branch and resubmit the PR. You can just copy and paste `z-bar.ts` file contents, and re-edit `index.ts` to export the plugin.

Thanks again!

**Pull request closed without merging**